test/test_alias.cpp

```diff
@@ -908,6 +908,27 @@ TEST_F(AliasTest, Broadcast) {
   EXPECT_EQ(out_tensor.data_ptr(), in_tensor.data_ptr());
 }
 
+TEST_F(AliasTest, MergeTwoExpandedBroadcasts) {
```

Comment:
Could you add a test that has both concrete and broadcasting dims? For example, contiguity
```
t n n t n n t
```
Because I am concerned about whether in the above code you can just `contiguity = true;`. 

For example, if a tensor has shape (4, 5, 6) and stride (6, 0, 1). Ideally, we should mark dim 1 as broadcasting, and have contiguity `(t, n, t)`. But if instead we are unable to mark dim 1 as broadcasting, we can not just replace the `n` as `t` to get `(t, t, t)`, instead, we should have `(f, f, t)`. Can this case be correctly handled?

Reply:
> Could you add a test that has both concrete and broadcasting dims?

Done and the test seems to work fine. 

> we can not just replace the `n` as `t` to get `(t, t, t)`

Can you clarify what would break? I'm also not quite confident about this change and feel there can be a problem.

Reply:
> Can you clarify what would break? I'm also not quite confident about this change and feel there can be a problem.

For example, if the tensor has shape (4, 5, 6) and stride (6, 0, 1), and the contiguity is `(t, t, t)`, then during lower, nvFuser will likely index the tensor like:
```python
for  i in range(4*5*6):
  tensor[0, 0, i]
```
which is wrong. The above indexing is correct only when the stride is `(30, 6, 1)`.

Reply:
Alas, I made a slightly more complicated test that breaks this... 

The root problem is that merging `e` is more restrictive than `t` **and even** `f`. For example, according to [this table](https://github.com/NVIDIA/Fuser/blob/0107a7ad24959b575662e94d699ebbc729a9e209/csrc/alias_analysis.cpp#L93-L98), `t,e=>C` but `t,f=>f` and `t,t`=>t. 

I'll think about this more. Chances are we'll have to properly fix #1126, @jacobhinkle